webtool/fetch.py

```
@@ -410,6 +415,14 @@ def _sprache_aus_env(roh, i: int):
 
 def download_one(project: str, url: str, sprecher=None, sprache=None) -> str:
     """Laedt die Tonspur nach projekte/<project>/audio/. Liefert den Basisnamen."""
+    # Namensraum-Riegel (#416), als ERSTE Vorbedingung — er steht ohne jede Wartezeit
+    # fest (billiger als ffmpeg und yt-dlp darunter): der CLI-Weg legt das Projekt
+    # selbst an, der Endpunkt prueft denselben Namen vorher. Vorbedingung statt
+    # Rohwurf (#173): kein pip, kein Wiederholungsdownload.
+    try:
+        paths.sicherer_projektname(project)
+    except ValueError as e:
+        raise Vorbedingung(f"Projektname unzulässig: {e}")
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)